testcases/OpTestSensors.py

```diff
@@ -0,0 +1,171 @@
+        cmd = "cat /boot/config-%s | grep -i SENSORS_IBMPOWERNV" % l_kernel
+        print cmd
+        try:
+            res = self.cv_LPAR._ssh_execute(cmd)
+            print res
+        except:
+            l_msg = "Getting Config file is failed"
+            print l_msg
+            raise OpTestError(l_msg)
+
+        try:
+            val = ((res.split("=")[1]).replace("\r\n", ""))
+            if val == "y":
+                print "Driver build into kernel itself"
+            else:
+                print "Driver will be built as module"
+        except:
+            print val
+            l_msg = "config option is not set,exiting..."
+            print l_msg
+            raise OpTestError(l_msg)
+
+        # Loading ibmpowernv driver only on powernv platform
+        self.cv_LPAR.lpar_load_ibmpowernv(l_oslevel)
+
+        # Checking for sensors command and lm_sensors package
+        response = self.cv_LPAR._ssh_execute("sensors")
+        matchObj = re.search(r"command not found", response)
+        if matchObj:
+            l_msg = "sensors not working, install lm_sensors package"
+            print l_msg
+            raise OpTestError(l_msg)
+        else:
+            pkg = self.cv_LPAR._ssh_execute("rpm -qf /bin/sensors")
+            print "Installed package:%s" % pkg
+        print response
+
+        try:
+            # Start the lm_sensors service
+            cmd = "/bin/systemctl stop  lm_sensors.service"
+            self.cv_LPAR._ssh_execute(cmd)
+            cmd = "/bin/systemctl start  lm_sensors.service"
+            self.cv_LPAR._ssh_execute(cmd)
+            cmd = "/bin/systemctl status  lm_sensors.service"
+            res = self.cv_LPAR._ssh_execute(cmd)
+            print res
+
+            # To detect different sensor chips and modules
+            res = self.cv_LPAR._ssh_execute("yes | sensors-detect")
+            print res
+        except:
+            l_msg = "loading lm_sensors service failed"
+            print l_msg
+            raise OpTestError(l_msg)
+
+        # Checking sensors command functionality with different options
+        output = self.cv_LPAR._ssh_execute("sensors; echo $?")
+        response = output.splitlines()
+        if int(response[-1]):
+            l_msg = "sensors not working,exiting...."
+            raise OpTestError(l_msg)
+        print output
+        output = self.cv_LPAR._ssh_execute("sensors -f; echo $?")
+        response = output.splitlines()
+        if int(response[-1]):
+            l_msg = "sensors -f not working,exiting...."
+            raise OpTestError(l_msg)
+        print output
+        output = self.cv_LPAR._ssh_execute("sensors -A; echo $?")
+        response = output.splitlines()
+        if int(response[-1]):
+            l_msg = "sensors -A not working,exiting...."
+            raise OpTestError(l_msg)
+        print output
+        output = self.cv_LPAR._ssh_execute("sensors -u; echo $?")
```

Comment:
Seems that in some cases, the _ssh_execute() is just going to be required for the test.  I'll leave it up to you to put what you think other tests can use into cv_LPAR interfaces, and the raw test commands can be done via the ._ssh_execute().